ext/storage_sources/gcp_store/gcp_connection.cpp

```diff
@@ -30,15 +30,35 @@
 
 #include <fstream>
 
-gcp_connection::gcp_connection(const std::string &bucket_name)
-    : _gcp_client(google::cloud::storage::Client()), _bucket_name(bucket_name)
+namespace gcs = google::cloud::storage;
+using namespace gcs;
+
+gcp_connection::gcp_connection(const std::string &bucket_name, const std::string &prefix)
+    : _gcp_client(google::cloud::storage::Client()), _bucket_name(bucket_name), _prefix(prefix)
 {
 }
 
 // Builds a list of object names from the bucket.
 int
-gcp_connection::list_objects(std::vector<std::string> &objects) const
+gcp_connection::list_objects(std::vector<std::string> &objects, bool list_single)
 {
+    /* Fetch the objects from the given bucket.
+     * If prefix is given ListObject will filter the objects in the bucket and only return the ones
+     * that match the prefix.
+     */
+    for (auto &&object_metadata : _gcp_client.ListObjects(_bucket_name, gcs::Prefix(_prefix))) {
+        /* Check if the current object is accessible (object exists but the user does not have
+        * permissions to access)
+        */
+        if (!object_metadata)
```

Comment:
If a user can't access an object I think we should do a bit more than just logging this info. For example exiting the function with an error code (`-1`) or not adding this object to the list.

My preference is exiting with an error code, but I'm not familiar with with our intended workflow. 
CC @jiechenbo

Reply:
We'll also want to update this `std::cerr` call once WT-10368 is delivered. Let's update the scope of WT-10368 to also remove any `std::err`/`std::cerr` calls we've added.

Reply:
We are planning to map return codes with familiar WT codes, so we will be eventually doing this, just not yet :)

Reply:
Sounds good! Can you confirm we have a ticket for this and then I'll resolve the comment 👍

Reply:
I added it into WT-10368 :)

Reply:
Thanks! Thread resolved